db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -3721,4 +3721,22 @@ void DBImpl::GetSnapshotContext(
   }
   *snapshot_seqs = snapshots_.GetAll(earliest_write_conflict_snapshot);
 }
+
+Status DBImpl::WaitForCompact(bool wait_unscheduled) {
+  // Wait until the compaction completes
+
+  // TODO: a bug here. This function actually does not necessarily
+  // wait for compact. It actually waits for scheduled compaction
+  // OR flush to finish.
+
+  InstrumentedMutexLock l(&mutex_);
+  while ((bg_bottom_compaction_scheduled_ || bg_compaction_scheduled_ ||
+          bg_flush_scheduled_ ||
+          (wait_unscheduled && unscheduled_compactions_)) &&
+         (error_handler_.GetBGError().ok())) {
+    bg_cv_.Wait();
+  }
+  return error_handler_.GetBGError();
+}
+
```

Comment:
Can the thing I added to db_bench be switched to using this? Not asking you to do that in this diff.
https://github.com/facebook/rocksdb/blob/main/tools/db_bench_tool.cc#L7788

Also, does this work for leveled, universal and integrated BlobDB? The thing I added to db_bench doesn't work for universal, see https://github.com/facebook/rocksdb/issues/9275

Reply:
> Can the thing I added to db_bench be switched to using this?

This is still an internal API which was used for unittest. To be used for db_bench, we need to expose this as a public API, which at least needs more polish and other considerations.

Reply:
@mdcallag looking at the code, I think you can do `dbimpl = static_cast<DBImpl*>(db_.db)`, but as @jay-zhuang pointed out, this is an internal function and may subject to change in the future.